### ISSUE TITLE
Remote image URIs without filename break HTML builds in 1.6.2

### ISSUE THREAD

**Issue by jantman** [sphinx-doc/sphinx#3860]
Subject: Remote image URIs without filename break HTML builds in 1.6.2

### Problem
- Remote image URIs that lack a filename, such as readthedocs.org status badges which are URIs of the form ``https://readthedocs.org/projects/PROJECT-NAME/badge/?version=latest`` (note there is no explicit filename here), break the HTML builder.

#### Procedure to reproduce the problem

Attempt building a project which includes an image like the above in a file, such as README.rst, using Sphinx 1.6.2. i.e.:

```
.. image:: https://readthedocs.org/projects/awslimitchecker/badge/?version=latest
   :target: https://readthedocs.org/projects/awslimitchecker/?badge=latest
   :alt: sphinx documentation for latest release
```

https://readthedocs.org/projects/awslimitchecker/badge/?version=latest actually redirects to https://img.shields.io/badge/docs-latest-brightgreen.svg?style=flat - the URL checks the status of the docs build for the specified version, and redirects to an appropriate shields.io badge.

Build with:

```
sphinx-build -a -n -W -b linkcheck docs/source docs/build/html
```

#### Error logs / results

```
Warning, treated as error:
Could not fetch remote image: https://readthedocs.org/projects/awslimitchecker/badge/?version=latest [[Errno 21] Is a directory: u'/home/jantman/GIT/awslimitchecker/docs/build/html/.doctrees/images/https/readthedocs.org/projects/awslimitchecker/badge//version=latest/']
```

- https://travis-ci.org/jantman/awslimitchecker/jobs/241720146

From what I can tell, the problem seems to be in [sphinx.transforms.post_transforms.images.ImageDownloader.handle](https://github.com/sphinx-doc/sphinx/blob/745e048e588f6661093dd8fc54440a1704a9dfcd/sphinx/transforms/post_transforms/images.py#L65) as introduced in #3589, as the local paths that it generates do not work for images without a filename (``basename``) in the URI.

#### Expected results

Expected these images to download correctly without a problem, as they did under 1.5.5

### Reproducible project / your project

- https://github.com/jantman/awslimitchecker

### Environment info
- OS: Linux (Arch, or whatever TravisCI and readthedocs.org run)
- Python version: 2.7.13
- Sphinx version: 1.6.2

**Comment by jfbu** [comment]
I can not reproduce your problem with minimal index.rst containing only 

```
Welcome to FOO's documentation!
===============================

.. image:: https://readthedocs.org/projects/awslimitchecker/badge/?version=latest
   :target: https://readthedocs.org/projects/awslimitchecker/?badge=latest
   :alt: sphinx documentation for latest release
```

and

```
# Sphinx version: 1.6.2
# Python version: 2.7.13 (CPython)
# Docutils version: 0.13.1 release
# Jinja2 version: 2.9.6
```

on mac os. Also tested with Python 3.5.3.

**Comment by jfbu** [comment]
apologies, yes indeed using the `sphinx-build` 

    $ sphinx-build -a -n -W -b linkcheck . _build/html

as you said. By the way on Sphinx 1.5.6, the same Warning is issued without even activating linkcheck.

**Comment by jfbu** [comment]
Well, sorry again, <strike>there is the problem even without linkcheck.</strike>

<strike>Error:</strike> No error with 1.6.2, but error with 1.5.6: 

    $ sphinx-build -a -n -W  . _build/html

No Error:

    $ make clean html O="-a -n -W"

Again Error:

    $ make clean html O="-b linkcheck -W"

Or simply one gets the warning via

     $ make clean linkcheck

Looks as if linkcheck needs update

**Comment by jantman** [comment]
Ok, thank you. Apologies for not posting a gist or repo, or some quicker way for you to reproduce the issue.

**Comment by tk0miya** [comment]
Fixed by #3890.
Thank you for reporting!
